dev/reference/best-practices/in-high-concurrency.md

```diff
@@ -0,0 +1,204 @@
+---
+title: TiDB 高并发写入场景最佳实践
+category: reference
+---
+
+# TiDB 高并发写入场景最佳实践
+
+在 TiDB 的使用过程中，一个典型场景是高并发批量写入数据到 TiDB。本文阐述了该场景中的常见问题，旨在给出一个业务的最佳实践，帮助读者在开发业务时避免陷入 TiDB 使用的 “反模式”。
+
+## 目标读者
+
+本文读者应该对 TiDB 有一定的了解，推荐先阅读 TiDB 原理的三篇文章（[讲存储](https://pingcap.com/blog-cn/tidb-internal-1/)，[说计算](https://pingcap.com/blog-cn/tidb-internal-2/)，[谈调度](https://pingcap.com/blog-cn/tidb-internal-3/)），以及 [TiDB Best Practice](https://pingcap.com/blog-cn/tidb-best-practice/)。
+
+## 高并发批量插入场景
+
+高并发批量插入的场景，通常产生在业务系统中的批量任务中，例如清算以及结算等业务。场景存在以下特点：
+
+- 数据量大
+- 需要短时间内将历史数据入库
+- 需要短时间内读取大量数据
+
+这就对 TiDB 提出了以下挑战：
+
+- 写入/读取能力是否可以线性水平扩展
+- 随着数据持续大并发写入，数据库性能是否稳定不衰减
+
+对于分布式数据库来说，除了本身的基础性能外，最重要的就是充分利用所有节点能力，避免让单个节点成为瓶颈。
+
+## TiDB 数据分布原理
+
+如果要解决以上挑战，需要从 TiDB 数据切分以及调度的原理开始讲起。这里只是作简单的说明，详情可参阅[谈调度](https://pingcap.com/blog-cn/tidb-internal-3/)。
+
+TiDB 以 Region 为单位对数据进行切分，每个 Region 有大小限制（默认 96M）。Region 的切分方式是范围切分。每个 Region 会有多副本，每一组副本，称为一个 Raft Group。每个 Raft Group 中由 Leader 负责执行这块数据的读 & 写（TiDB 即将支持 [Follower-Read](https://zhuanlan.zhihu.com/p/78164196)）。Leader 会自动地被 PD 组件均匀调度在不同的物理节点上，用以均分读写压力。
+
+![TiDB 数据概览](/media/high-concurrency-best-practice/tidb-data-overview.png)
+
+只要业务的写入没有 AUTO_INCREMENT 的主键，或没有单调递增的索引（即没有业务上的写入热点，更多细节可参阅 [TiDB 正确使用方式](https://zhuanlan.zhihu.com/p/25574778)），从原理上来说，TiDB 依靠这个架构可具备线性扩展的读写能力，并且可以充分利用分布式资源。从这一点看，TiDB 尤其适合高并发批量写入场景的业务。
+
+但理论场景和实际情况往往存在不同。以下实例说明了热点是如何产生的。
+
+## 热点产生的实例
+
+以下为一张示例表：
+
+```sql
+CREATE TABLE IF NOT EXISTS TEST_HOTSPOT(
+      id         BIGINT PRIMARY KEY,
+      age        INT,
+      user_name  VARCHAR(32),
+      email      VARCHAR(128)
+)
+```
+
+这个表的结构非常简单，除了 `id` 为主键以外，没有额外的二级索引。将数据写入该表的语句如下，`id` 通过随机数离散生成：
+
+{{< copyable "sql" >}}
+
+```sql
+INSERT INTO TEST_HOTSPOT(id, age, user_name, email) values(%v, %v, '%v', '%v');
+```
+
+负载是短时间内密集地执行以上写入语句。
+
+以上操作看似符合理论场景中的 TiDB 最佳实践，业务上没有热点产生。只要有足够的机器，就可以充分利用 TiDB 的分布式能力。要验证是否真的符合最佳实践，可以在实验环境中进行测试。
+
+部署拓扑 2 个 TiDB 节点，3 个 PD 节点，6 个 TiKV 节点。请忽略 QPS，因为测试只是为了阐述原理，并非 benchmark。
+
+![QPS1](/media/high-concurrency-best-practice/QPS1.png)
+
+客户端在短时间内发起了 “密集” 的写入，TiDB 收到的请求是 3K QPS。理论上，压力应该均摊给 6 个 TiKV 节点。但是从 TiKV 节点的 CPU 使用情况上看，存在明显的写入倾斜（tikv - 3 节点是写入热点）：
```

Comment:
```suggestion
客户端在短时间内发起了“密集”的写入，TiDB 收到的请求是 3K QPS。理论上，压力应该均摊给 6 个 TiKV 节点。但是从 TiKV 节点的 CPU 使用情况上看，存在明显的写入倾斜（tikv - 3 节点是写入热点）：
```

Reply:
Do not leave extra spaces around Chinese punctuation. Please get familiar with our style guide.